File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_04_二叉搜索树属性/LeetCode_0530.java
//530. 二叉搜索树的最小绝对差
package com.kurumi.binarytree._04_二叉搜索树属性;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0530 {

}
class Solution {
  TreeNode pre=null;
  int res=Integer.MAX_VALUE;
  public void traversal(TreeNode root){
    if(root==null) return;
    traversal(root.left);
    if(pre!=null) res=Integer.min(res, root.val-pre.val);
    pre=root;
    traversal(root.right);
  }
  public int getMinimumDifference(TreeNode root) {
    traversal(root);
    return res;
  }
}

File: 卡码/代码随想录/Python/二叉树/04-二叉搜索树属性/LeetCode_0530.py
#530. 二叉搜索树的最小绝对差
import sys
from typing import *
from ..TreeNode import *

class Solution:
  def __init__(self):
    self.res:int=sys.maxsize
    self.pre:TreeNode=None
  def traversal(self,root:Optional[TreeNode]):
    if not root:
      return
    self.traversal(root.left)
    if self.pre:
      self.res=min(self.res,root.val-self.pre.val)
    self.pre=root
    self.traversal(root.right)
  def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
    self.traversal(root)
    return self.res
